pennylane/gradients/finite_difference.py

```diff
@@ -473,7 +463,7 @@ def _single_shot_batch_result(results):
         return tuple(tuple(elem) for elem in grads_reorder)
 
     processing_fn = functools.partial(
-        _processing_fn, shots=shots, single_shot_batch_fn=_single_shot_batch_result
+        _processing_fn, shots=tape.shots, single_shot_batch_fn=_single_shot_batch_result

```

Comment:
[Commenting here because the definition line of `_processing_fn` wasn't touched in the PR and can't be commented]

I think we could move to have `_processing_fn` only take positional arguments, because it is a very particular function only used once here and once in the spsa gradient, at the moment.
Then we can get rid of the dummy `shots=Shots(None)` which is executed whenever loading PL, as far as I am aware.

Reply:
done!